# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [1]:
import pandas as pd
cars = pd.read_csv('../DATA/car-sales.csv')
cars.head(5)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0

In [21]:
# Your code here
import pandas as pd
cars = pd.read_csv('../DATA/car-sales.csv')

print(f"Số dòng là : {cars.shape[0]}")
print(f"Số cột là : {cars.shape[1]}")
cars.shape


Số dòng là : 10
Số cột là : 5


(10, 5)

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [8]:
# Your code here
# cars.describe()
cars.describe(include='all')


# Thuộc tính phân loại gồm : Manufacturer, Colour
# Thuộc tính định lượng gồm : Odometer, Doors, Price

Manufacturer Colour       Odometer      Doors         Price
count            10     10      10.000000  10.000000     10.000000
unique            4      5            NaN        NaN           NaN
top          Toyota  White            NaN        NaN           NaN
freq              4      4            NaN        NaN           NaN
mean            NaN    NaN   78601.400000   4.000000   7645.000000
std             NaN    NaN   61983.471735   0.471405   5379.407753
min             NaN    NaN   11179.000000   3.000000   3500.000000
25%             NaN    NaN   35836.250000   4.000000   4625.000000
50%             NaN    NaN   57369.000000   4.000000   6625.000000
75%             NaN    NaN   96384.500000   4.000000   7375.000000
max             NaN    NaN  213095.000000   5.000000  22000.000000

### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [27]:
# Your code here
cars.Manufacturer.count()
print(f"{cars.Manufacturer.count()} xe")

10 xe


### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 4 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [29]:
# Your code here
cars.groupby(['Manufacturer']).size()

Manufacturer
BMW       1
Honda     3
Nissan    2
Toyota    4
dtype: int64

### 5. Đếm số xe theo màu


In [66]:
# Your code here
cars.groupby(['Colour']).size()

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [72]:
# Your code here
sum = cars.Price.sum()
print(f"Tổng giá trị các xe trong cửa hàng là {sum} USD")

76450.0

### 7. Top 3 xe có giá cao nhất


In [44]:
# Your code here
#Sắp xếp tăng dần theo giá. Sau đó lấy 3 xe đầu tiên
cars[["Manufacturer", 'Colour', 'Price']].sort_values(by = 'Price', ascending = False).head(3)



Manufacturer Colour    Price
3          BMW  Black  22000.0
9       Nissan  White   9700.0
6        Honda   Blue   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [69]:
# Your code here
cars[["Manufacturer", 'Colour', 'Odometer']].sort_values(by = 'Odometer', ascending = False).head(3)

Manufacturer Colour  Odometer
4       Nissan  White    213095
0       Toyota  White    150043
5       Toyota  Green     99213

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [78]:
# Your code here
car_Honda = cars[cars['Manufacturer'] == 'Honda']
# car_Honda.shape[1]
print(f"Tỷ lệ xe Honda trong dữ liệu của cửa hàng là : {car_Honda.shape[0]/cars.shape[0]}")
print(f"Giá trị trung bình của xe hãng Honda của cửa hàng là : {int(car_Honda.Price.mean())}")
print(f"Các xe của hãng Honda trong cửa hàng có những màu sắc là : {car_Honda['Colour'].unique()}")



Tỷ lệ xe Honda trong dữ liệu của cửa hàng là : 0.3
Giá trị trung bình của xe hãng Honda của cửa hàng là : 6500
Các xe của hãng Honda trong cửa hàng có những màu sắc là : ['Red' 'Blue']


### 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [9]:
# Giá trị trung bình các xe
cars['Price'].mean()

7645.0

In [10]:
# Giá trị trung vị các xe
cars['Price'].median()

6625.0

In [11]:
# Miền giá trị của xe : max - min
cars['Price'].max() - cars['Price'].min()

18500.0

In [12]:
# Miền phân vị của xe : quantile(0.75) - quantile(0.25)
cars['Price'].quantile(0.75) - cars['Price'].quantile(0.25)

2750.0

In [15]:
# Điểm phân vị
print(cars['Price'].quantile(0.4))
print(cars['Price'].quantile(0.6))

5750.0
7000.0


---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [19]:
# Your code here
tips = pd.read_csv('../DATA/tips.csv')
tips.head(10)

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4
5           5       25.29  4.71    Male     No  Sun  Dinner     4
6           6        8.77  2.00    Male     No  Sun  Dinner     2
7           7       26.88  3.12    Male     No  Sun  Dinner     4
8           8       15.04  1.96    Male     No  Sun  Dinner     2
9           9       14.78  3.23    Male     No  Sun  Dinner     2

2. Xoá cột `Unnamed: 0`


In [21]:
# Your code here
tips.drop(columns=['Unnamed: 0'])

total_bill   tip     sex smoker   day    time  size
0         16.99  1.01  Female     No   Sun  Dinner     2
1         10.34  1.66    Male     No   Sun  Dinner     3
2         21.01  3.50    Male     No   Sun  Dinner     3
3         23.68  3.31    Male     No   Sun  Dinner     2
4         24.59  3.61  Female     No   Sun  Dinner     4
..          ...   ...     ...    ...   ...     ...   ...
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

[244 rows x 7 columns]

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [34]:
# Your code here
tips.shape

(244, 8)

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [48]:
# Your code here
print(tips.dtypes)


Unnamed: 0      int64
total_bill    float64
tip           float64
sex            object
smoker         object
day            object
time           object
size            int64
dtype: object


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [60]:
# Your code here
sex_counts = tips['sex'].value_counts()
print("Số lượng nam :",sex_counts['Male'])
print("Số lượng nữ :",sex_counts['Female'])

print("Tỉ lệ nam :", sex_counts['Male']/(len(tips) - 1))
print("Tỉ lệ nữ :", sex_counts['Female']/(len(tips) - 1))


Số lượng nam : 157
Số lượng nữ : 87
Tỉ lệ nam : 0.6460905349794238
Tỉ lệ nữ : 0.35802469135802467


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [66]:
# Your code here
smoker_counts = tips['smoker'].value_counts()
print("Số người hút thuốc là :", smoker_counts['Yes'])
print("Tỷ lệ :", smoker_counts['Yes']/(len(tips) - 1))

Số người hút thuốc là : 93
Tỷ lệ : 0.38271604938271603


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?


In [76]:
# Your code here
day_counts = tips['day'].value_counts()
print("Ngày trong tuần khách hàng hay đến dùng bữa nhất là", day_counts.idxmax())
print("Tỉ lệ", day_counts[day_counts.idxmax()] / (len(tips) - 1))


Ngày trong tuần khách hàng hay đến dùng bữa nhất là Sat
Tỉ lệ 0.35802469135802467


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?


In [77]:
# Your code here
time_counts = tips['time'].value_counts()
print("Nhà hàng phục vụ nhiều nhất là bữa ăn", time_counts.idxmax())

Nhà hàng phục vụ nhiều nhất là bữa ăn Dinner


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [81]:
# Your code here
size_max = tips['size'].max()
print("Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là", size_max, "người")

Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là 6 người


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [84]:
# Your code here
print("Trung bình trong một bữa ăn nhà hàng phục vụ", int(tips['size'].mean()),"khách")

Trung bình trong một bữa ăn nhà hàng phục vụ 2 khách


---